shared/profile/user/weboftrust/index.tsx

```diff
@@ -1,68 +1,106 @@
+            style={styles.avatar}
+            showFollowingStatus={false}
+          />
+        </Kb.Box2>
+        <Kb.Box2 direction="vertical" fullWidth={true} style={styles.textContainer} centerChildren={false}>
+          <Kb.Text style={styles.attestationText} type="Body">
+            {props.attestation}
+          </Kb.Text>
+          <Kb.Box2 direction="vertical" style={styles.signatureBox} centerChildren={false} fullWidth={true}>
+            <Kb.Box2
+              direction="horizontal"
+              gap="xxtiny"
+              centerChildren={true}
+              style={{alignSelf: 'flex-start'}}
+            >
+              <Kb.Icon color={Styles.globalColors.blue} type="iconfont-proof-good" sizeType="Small" />
+              <Kb.Text type="BodySmall">signed by </Kb.Text>
+              <Kb.ConnectedUsernames
+                type={Styles.isMobile ? 'BodySmallBold' : 'BodyBold'}
+                usernames={props.attestingUser}
+                colorBroken={true}
+                colorFollowing={true}
+                style={styles.username}
+              />
+              <Kb.Text type="BodySmall">{formatTimeRelativeToNow(props.vouchedAt)}</Kb.Text>
+            </Kb.Box2>
           </Kb.Box2>
         </Kb.Box2>
       </Kb.Box2>
-    </Kb.Box2>
-    {(props.onHide || props.onAccept || props.onReject) && (
-      <Kb.Box2 direction="horizontal" fullWidth={true} centerChildren={false} style={styles.buttonBar}>
-        <Kb.ButtonBar align="flex-start">
-          {props.onHide && <Kb.Button label="Hide" small={true} type="Danger" onClick={props.onHide} />}
-          {props.onAccept && (
-            <Kb.Button label="Accept" small={true} type="Success" onClick={props.onAccept} />
-          )}
-          {props.onReject && <Kb.Button label="Reject" small={true} type="Danger" onClick={props.onReject} />}
-        </Kb.ButtonBar>
-      </Kb.Box2>
-    )}
-  </>
-)
+      {(props.onHide || props.onAccept || props.onReject) && (
+        <Kb.Box2 direction="horizontal" fullWidth={true} centerChildren={false} style={styles.buttonBar}>
+          <Kb.ButtonBar align="flex-start">
+            {props.onHide && <Kb.Button label="Hide" small={true} type="Danger" onClick={props.onHide} />}
+            {props.onAccept && (
+              <Kb.WaitingButton
+                label="Accept"
+                small={true}
+                type="Success"
+                onClick={props.onAccept}
+                waitingKey={props.reactWaitingKey}
+              />
+            )}
+            {props.onReject && (
+              <Kb.WaitingButton
+                label="Reject"
+                small={true}
+                type="Danger"
+                onClick={props.onReject}
+                waitingKey={props.reactWaitingKey}
+              />
+            )}
+          </Kb.ButtonBar>
+        </Kb.Box2>
+      )}
+    </>
+  )
+}
+
+const WebOfTrust = (props: Props) => {
+  switch (props.status) {
```

Comment:
hm, something feels wrong if we're getting all the way here in many circumstances where we might just return null.  Seems like the container should take care of not trying to render null rows, and the container above that one should take care of not even showing the "Web of Trust (0)" tab if we know there's nothing to render?  Maybe you could explain this part more.

Reply:
yeah - that makes sense. it's not actually obvious to me where all of this logic _should_ live considering how much context you need to decide whether or not to show each entry. it's not a crazy amount, but it's some work. and i kind of balked at doing all of it in this PR because it started looking surprisingly complicated, and i thought it warranted it's own effort after all of this plumbing/wiring was in (here i was mainly just focused on getting the accept/reject buttons to work). that ticket is here https://keybase.atlassian.net/browse/PICNIC-1062

i can do some of it now, but i don't think it will be missed since it's fairly explicitly ticketed.